packages/webview_flutter/webview_flutter_android/android/src/test/java/io/flutter/plugins/webviewflutter/WebViewTest.java

```diff
@@ -14,343 +14,335 @@
   }
 
   @Test
-  public void defaultWebChromeClientDoesNotAttemptToCommunicateWithDart() {
-    final WebViewPlatformView webView = new WebViewPlatformView(mockContext, null, null);
-    // This shouldn't throw an Exception.
-    Objects.requireNonNull(webView.getWebChromeClient()).onProgressChanged(webView, 0);
-  }
+  public void setBackgroundColor() {
+    final PigeonApiWebView api = new TestProxyApiRegistrar().getPigeonApiWebView();
 
-  @Test
-  public void disposeDoesNotCallDestroy() {
-    final boolean[] destroyCalled = {false};
-    final WebViewPlatformView webView =
-        new WebViewPlatformView(mockContext, null, null) {
-          @Override
-          public void destroy() {
-            destroyCalled[0] = true;
-          }
-        };
-    webView.dispose();
+    final WebView instance = mock(WebView.class);
+    final long color = 0L;
+    api.setBackgroundColor(instance, color);
 
-    assertFalse(destroyCalled[0]);
+    verify(instance).setBackgroundColor((int) color);
   }
 
   @Test
-  public void destroyWebViewWhenDisposedFromJavaObjectHostApi() {
-    final boolean[] destroyCalled = {false};
-    final WebViewPlatformView webView =
-        new WebViewPlatformView(mockContext, null, null) {
-          @Override
-          public void destroy() {
-            destroyCalled[0] = true;
-          }
-        };
-
-    testInstanceManager.addDartCreatedInstance(webView, 1);
-    final JavaObjectHostApiImpl javaObjectHostApi = new JavaObjectHostApiImpl(testInstanceManager);
-    javaObjectHostApi.dispose(1L);
+  public void defaultWebChromeClientIsSecureWebChromeClient() {
+    final WebViewProxyApi mockApi = mock(WebViewProxyApi.class);
+    when(mockApi.getPigeonRegistrar()).thenReturn(new TestProxyApiRegistrar());
+    final WebViewProxyApi.WebViewPlatformView webView =
+        new WebViewProxyApi.WebViewPlatformView(mockApi);
 
-    assertTrue(destroyCalled[0]);
+    assertTrue(
+        webView.getWebChromeClient() instanceof WebChromeClientProxyApi.SecureWebChromeClient);
+    assertFalse(
+        webView.getWebChromeClient() instanceof WebChromeClientProxyApi.WebChromeClientImpl);
   }
 
+  // This test verifies that WebView.destroy() is called when the Dart instance is garbage collected.
+  // This requires adding
+  //
+  // ```
+  // val instance: Any? = getInstance(identifier)
+  // if (instance is WebViewPlatformView) {
+  //   instance.destroy()
+  // }
+  // ```
+  //
+  // to `AndroidWebkitLibraryPigeonInstanceManager.remove` in the generated code. This is done as a
+  // temporary workaround to prevent the transition to the new pigeon ProxyApi generator from being
+  // a breaking change. Maintainers should consider whether continuing  to call `destroy` on
+  // `WebView` is valuable.
   @Test
-  public void flutterApiCreate() {
-    final InstanceManager instanceManager = InstanceManager.create(identifier -> {});
+  public void destroyWebViewWhenRemovedFromInstanceManager() {
```

Comment:
Drawing attention to this test to get feedback. See comment above.

Reply:
Yea it's the timing of the removal of the object. The `InstanceManager` can't be overridden since it is final and attempting to add another `Finalizer` to the Dart instance could cause a race condition.

  I was thinking about this a bit more and I think it would be pretty reasonable to just add an `onDartInstanceRemoved` to the generated native `InstanceManger` in a `pigeon` PR. What are your thoughts?